rts/lra/lra-test/basic/src/test/java/io/narayana/lra/arquillian/extension/ConfigAuxiliaryArchiveAppender.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2019, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package io.narayana.lra.arquillian.extension;
+
+import io.narayana.lra.archive.LraArchiveProvider;
+import org.jboss.arquillian.container.test.spi.client.deployment.AuxiliaryArchiveAppender;
+import org.jboss.shrinkwrap.api.Archive;
+
+/**
+ * The appender provides way to bundle LRA interfaces implementation classes to the deployment
+ * as the Thorntail container does not bundle the LRA classes to the fat jar until
+ * there is no fraction for it.
+ */
+public class ConfigAuxiliaryArchiveAppender implements AuxiliaryArchiveAppender {
+
+    @Override
+    public Archive<?> createAuxiliaryArchive() {
+        return LraArchiveProvider.createLraArchive();
```

Comment:
would not be better to just used the `LRAArchiveProvider` as an extension then redirects here the call from here to a different class? I mean to define the extension at `tck/src/test/resources/META-INF/services/org.jboss.arquillian.core.spi.LoadableExtension` and remove this class completely?

Reply:
I tried and it didn't work - it wasn't picked by arquillian. I can try again I think if you want.